src/components/ToolBar.vue

```diff
@@ -1,113 +1,67 @@
+import SubToolMenu from '@/components/SubToolMenu.vue'
 
-export type Color = {
-  color: string
-  style: {
-    'background-color'?: string
-    background?: string
-  }
-  onClick: () => void
-}
-
-const { tool, drawColor } = storeToRefs(useStoreLine())
-const { setTool, setColor, setGlobalCompositeOperation } = useStoreLine()
-const { changeToPreparedColor } = useStoreColor()
-
-const paletteIsOpen = ref(false)
-const activeIndex = ref<number | null>(null)
-
-const colors: Color[] = [
-  {
-    color: '#1E1E1E',
-    style: { 'background-color': '#1E1E1E' },
-    onClick: () => setColor('#1E1E1E'),
-  },
-  {
-    color: '#F24822',
-    style: { 'background-color': '#F24822' },
-    onClick: () => setColor('#F24822'),
-  },
-  {
-    color: '#FFA629',
-    style: { 'background-color': '#FFA629' },
-    onClick: () => setColor('#FFA629'),
-  },
-  {
-    color: '#FFCD29',
-    style: { 'background-color': '#FFCD29' },
-    onClick: () => setColor('#FFCD29'),
-  },
-  {
-    color: '#14AE5C',
-    style: { 'background-color': '#14AE5C' },
-    onClick: () => setColor('#14AE5C'),
-  },
-  {
-    color: '#0D99FF',
-    style: { 'background-color': '#0D99FF' },
-    onClick: () => setColor('#0D99FF'),
-  },
-  {
-    color: '#9747FF',
-    style: { 'background-color': '#9747FF' },
-    onClick: () => setColor('#9747FF'),
-  },
-  {
-    color: '#FFFFFF',
-    style: { 'background-color': '#FFFFFF' },
-    onClick: () => setColor('#FFFFFF'),
-  },
-  {
-    color: 'rainbow',
-    style: {
-      background: 'conic-gradient(red, yellow, lime, aqua, blue, magenta, red)',
-    },
-    onClick: () => {
-      paletteIsOpen.value = !paletteIsOpen.value
-      // 色をカラーピッカーにも同期させる
-      changeToPreparedColor(drawColor.value)
-    },
-  },
-]
+const { mode } = storeToRefs(useStoreMode())
+const { setMode } = useStoreMode()
+const { setLineStyle, setGlobalCompositeOperation } = useStoreLine()
 </script>
 
 <template lang="pug">
-
-.flex.justify-center 
-  div(class="flex justify-center items-center p-6 bg-white rounded-lg border border-gray-200 shadow-md dark:bg-gray-800 dark:border-gray-700 mt-3")
+div(class="flex flex-col items-center absolute bottom-2 left-1/2 -translate-x-1/2")
+  SubToolMenu
+  div(class="flex justify-center items-center px-4 py-3 bg-white rounded-lg border border-gray-200 shadow-md dark:bg-gray-800 dark:border-gray-700 mx-auto")
     ul.flex
+      //- Select
+      li.flex.mx-2
+        button(v-show="mode !== 'select'" class="bg-transparent hover:bg-blue-500 text-blue-700 font-semibold hover:text-white py-2 px-4 border border-blue-500 hover:border-transparent rounded" @click="() => {setMode('select');}")
+          span(class="material-symbols-outlined") navigation
+        button(v-show="mode === 'select'" class="bg-blue-500 hover:bg-blue-500 font-semibold text-white py-2 px-4 border border-blue-500 hover:border-transparent rounded" @click="setMode('none')")
+          span(class="material-symbols-outlined") navigation
```

Comment:
```suggestion
     li(v-for='(arr, index) of array' :key='index' class='flex mx-2')
        button(v-show="mode !== arr.show" class="bg-transparent hover:bg-blue-500 text-blue-700 font-semibold hover:text-white py-2 px-4 border border-blue-500 hover:border-transparent rounded" @click="arr.event")
          span(class="material-symbols-outlined") {{arr.text}}
        button(v-show="mode === arr.show" class="bg-blue-500 hover:bg-blue-500 font-semibold text-white py-2 px-4 border border-blue-500 hover:border-transparent rounded" @click="setMode('none')")
          span(class="material-symbols-outlined") {{arr.text}}
```
```ts
const array: {
  text: string;
  show: string;
  event: () => void;
}[] = reactive([
  {
    text: "navigation",
    show: "select",
    event: () => {
      setMode("select");
    },
  },
  {
    text: "edit",
    show: "pen",
    event: () => {
      setMode("pen");
      setGlobalCompositeOperation("pen");
    },
  },
]);
```
変数名適当ですが、htmlに書くより配列に格納した方が追加・修正がしやすいと思います。

Reply:
なるほど、そうやって書けばよかったんですね！
ありがとうございます。すごくすっきりしました！